drivers/flash/soc_flash_silabs_series2.c

```diff
@@ -75,9 +93,51 @@ static int flash_silabs_read(const struct device *dev, off_t offset, void *data,
 		return 0;
 	}
 
+#if defined(CONFIG_SOC_FLASH_SILABS_S2_DMA_READ)
+	struct flash_silabs_data *const dev_data = dev->data;
+	size_t data_size = 1;
+	int err;
+
+	if (IS_ALIGNED(offset, sizeof(uint32_t)) && IS_ALIGNED(data, sizeof(uint32_t)) &&
+	    IS_ALIGNED(size, 4)) {
+		data_size = 4;
+	}
+
+	dev_data->dma_cfg.source_data_size = data_size;
+	dev_data->dma_cfg.dest_data_size = data_size;
+	dev_data->dma_cfg.source_burst_length = data_size;
+	dev_data->dma_cfg.dest_burst_length = data_size;
+
+	dev_data->dma_block_cfg = (struct dma_block_config){
+		.block_size = size,
+		.source_address = CONFIG_FLASH_BASE_ADDRESS + offset,
+		.dest_address = (uint32_t)data,
+	};
+
+	k_sem_take(&dev_data->mutex, K_FOREVER);
```

Comment:
It is probably a good idea to include accesses to `dev_data->dma_cfg` and `dev_data->dma_block_cfg` in this critical section.